### DISCUSSION TITLE
How to display an app in the user admin current language?

### DISCUSSION THREAD

**Discussion by JumBay** [Shopify/polaris-react#8300] [Q&A] [answered]
Hi,

I don't know if this is the right place to ask, but I was wondering, how to get the user's locale to display an embedded app with the same language as its admin? Maybe it's more related to app bridge, but I can't find any answer

**Comment by chloerice** [comment]
Hey @JumBay 👋🏽 You can use the [Admin GraphQL API](https://shopify.dev/docs/api/admin-graphql/2023-01/objects/ShopLocale) to query for shop's locale details, including the default locale.

**Reply by JumBay** [reply]
Thanks for the reply, I've already tried this, but it doesn't return the user admin locale but the language set in settings > language.
I've managed to find the user locale from:
```ts
import { createApp } from "@shopify/app-bridge";

const app = createApp(....);
const appState = await app.getState();
const locale = appState?.staffMember?.locale 
```
But It don't know if it's the right way to do so.

**Comment by chloerice** [comment, marked as answer]
> Thanks for the reply, I've already tried this, but it doesn't return the user admin locale but the language set in settings > language. I've managed to find the user locale from:
> 
> ```ts
> import { createApp } from "@shopify/app-bridge";
> 
> const app = createApp(....);
> const appState = await app.getState();
> const locale = appState?.staffMember?.locale 
> ```
> 
> But It don't know if it's the right way to do so.

Yes, I just asked the team that maintains App Bridge and they've confirmed using the `appState.staffMember.locale` is the correct way to get the current admin user's locale 😺 Thank you for sharing, I didn't know about this!

**Reply by JumBay** [reply]
Thanks. I've also found out that embedded app can retrieve the locale like this too:
```ts
new URLSearchParams(location.search).get("locale")
```
But it's not documented.
